docs/ops/production_ready.md

```diff
@@ -22,76 +22,54 @@ specific language governing permissions and limitations
+To establish a stable mapping, we need stable operator uids provided by the user through `setUid(String uid)`.
 
-### Choice of state backend
+### Choose The Right State Backend
 
-Currently, Flink has the limitation that it can only restore the state from a savepoint for the same state backend that
-took the savepoint. For example, this means that we can not take a savepoint with a memory state backend, then change
-the job to use a RocksDB state backend and restore. While we are planning to make backends interoperable in the near
-future, they are not yet. This means you should carefully consider which backend you use for your job before going to
-production.
+Currently, Flink's savepoint binary format is state backend specific. 
+This means that a savepoint taken with one state backend cannot be restored using another and you should carefully consider which backend you use before going to production.
 
-In general, we recommend using RocksDB because this is currently the only state backend that supports large states (i.e.
-state that exceeds the available main memory) and asynchronous snapshots. From our experience, asynchronous snapshots are
-very important for large states because they do not block the operators and Flink can write the snapshots without stopping 
-stream processing. However, RocksDB can have worse performance than, for example, the memory-based state backends. If
-you are sure that your state will never exceed main memory and blocking the stream processing to write it is not an issue,
-you **could consider** to not use the RocksDB backends. However, at this point, we **strongly recommend** using RocksDB
-for production.
+In general, we recommend avoiding `MemoryStateBackend` in production because it stores its snapshots inside the JobManager as opposed to persistent disk. 
+When deciding between `FsStateBackend` and `RocksDB` it is a choice between performance and scalability. 
+`FsStateBackend` is very fast as each state access and update is simply passing pointers to objects on the Java heap, however, it is limited by available memory within the cluster. 
+On the other hand, `RocksDB` can scale based on available disk space and is the only state backend to support incremental snapshots. 
+However, each state access and update requires (de-)serialization and potentially reading from disk which leads to average performance that is an order of magnitude slower than the memory state backends. 
+Carefully read through the [state backend documentation]({{ site.baseurl }}/ops/state/state_backends.html) to fully understand the pros and cons of each option. 
 
-### Config JobManager High Availability(HA)
+### Configure JobManager High Availability
 
-The JobManager coordinates every Flink deployment. It is responsible for both *scheduling* and *resource management*.
+The JobManager serves as a central coordinator for each Flink deployment, being responsible for both *scheduling* and *resource management*.
```

Comment:
I think this section is still a bit confusing. It makes no difference if you have a single JobManager or multiple JobManagers, as long as that single JobManager process comes back (which it does on Yarn, K8s, Mesos, etc.).

The issue is that without ZK, the JobManager process cannot recover once it comes back, and neither can the other standby JobManagers in the case of multiple JMs.